src/osiris_log.erl

```diff
@@ -1251,57 +1275,96 @@ parse_records(Offs,
     parse_records(Offs + NumRecs, Rem, lists:reverse(Recs) ++ Acc).
 
 build_log_overview(Dir) when is_list(Dir) ->
+  build_log_overview(Dir, []).
+
+build_log_overview(Dir, Opts) when is_list(Dir) ->
     {Time, Result} = timer:tc(
                        fun() ->
                                try
                                    IdxFiles =
                                    lists:sort(
                                      filelib:wildcard(
                                        filename:join(Dir, "*.index"))),
-                                   build_log_overview0(IdxFiles, [])
+                                   build_log_overview0(IdxFiles, Opts, [])
                                catch
                                    missing_file ->
-                                       build_log_overview(Dir)
+                                       build_log_overview(Dir, Opts)
                                end
                        end),
     ?DEBUG("~s:~s/~b completed in ~fs", [?MODULE, ?FUNCTION_NAME, ?FUNCTION_ARITY, Time/1000000]),
     Result.
 
-build_log_overview0([], Acc) ->
+build_log_overview0([], _Opts, Acc) ->
     lists:reverse(Acc);
-build_log_overview0([IdxFile | IdxFiles], Acc0) ->
-    IdxFd = open_index_read(IdxFile),
-    case file:position(IdxFd, {eof, -?INDEX_RECORD_SIZE_B}) of
-        {error, einval} when IdxFiles == [] andalso Acc0 == [] ->
-            %% this would happen if the file only contained a header
-            ok = file:close(IdxFd),
-            SegFile = segment_from_index_file(IdxFile),
-            [#seg_info{file = SegFile, index = IdxFile}];
-        {error, einval} ->
-            ok = file:close(IdxFd),
-            build_log_overview0(IdxFiles, Acc0);
-        {ok, Pos} ->
-            %% ASSERTION: ensure we don't have rubbish data at end of idex
-            0 = (Pos - ?IDX_HEADER_SIZE) rem ?INDEX_RECORD_SIZE_B,
-            case file:read(IdxFd, ?INDEX_RECORD_SIZE_B) of
-                {ok,
-                 <<_Offset:64/unsigned,
-                   _Timestamp:64/signed,
-                   _Epoch:64/unsigned,
-                   LastChunkPos:32/unsigned>>} ->
-                    ok = file:close(IdxFd),
+build_log_overview0([IdxFile | IdxFiles], Opts, Acc0) ->
```

Comment:
@ansd There is the `rebar3 format` command to format the code. You may want to use it after the PR is merged to avoid embedding the formatting of previous "un-formatted" commits in your changes.